src/usermod.c

```diff
@@ -1073,18 +1073,14 @@ static void process_flags (int argc, char **argv)
 				}
 				break;
 			case 'e':
-				if ('\0' != *optarg) {
-					user_newexpire = strtoday (optarg);
-					if (user_newexpire < -1) {
-						fprintf (stderr,
-						         _("%s: invalid date '%s'\n"),
-						         Prog, optarg);
-						exit (E_BAD_ARG);
-					}
-					user_newexpire *= DAY / SCALE;
-				} else {
-					user_newexpire = -1;
+				user_newexpire = strtoday (optarg);
+				if (user_newexpire < -1) {
+					fprintf (stderr,
+						 _("%s: invalid date '%s'\n"),
+						 Prog, optarg);
+					exit (E_BAD_ARG);
 				}
+				user_newexpire *= DAY / SCALE;
```

Comment:
You are correct that the '\0' case is handled in strtoday.  However, here you will take -1 and set user_newexpire = -1 * DAY / SCALE.

Reply:
Yes, I thought about that, but then this result is in the end passed to `date_to_str()`, isn't it?  And it interprets any negative number as `"never"`, right?

https://github.com/shadow-maint/shadow/blob/master/src/usermod.c#L1109

https://github.com/shadow-maint/shadow/blob/master/src/usermod.c#L631

https://github.com/shadow-maint/shadow/blob/master/src/usermod.c#L216

Reply:
However, having the special case might make it more explicit.

Reply:
My ctags file has let me to the chage.c version of date_to_str.  Which
still might have ended up ok but was less obvious.

This'll do - thank you.

Reply:
Maybe `date_to_str()` could be moved to `<libmisc/>` to have a single definition?

Reply:
> Maybe `date_to_str()` could be moved to `<libmisc/>` to have a single definition?

Looks like in each case it's to do with expiration, so I think that's a good idea.  Do you want to propose a patch?

Reply:
Sure :)

Reply:
What's the difference between `<lib/>` and `<libmisc/>`?  BTW, I also found a definition of `date_to_str()` in `<src/passwd.c>`, so I'll consider that one too.  In that one, I'll move the static buffer to a global static one to make it work with the new definition.

Reply:
Patch proposed: https://github.com/shadow-maint/shadow/pull/467